06_puppy/mohankrishna/map_store.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+type Puppy struct {
+	ID     uint32
+	Breed  string
+	Colour string
+	Value  int
+}
+
+type Storer interface {
+	CreatePuppy(puppy *Puppy)
+	ReadPuppy(ID uint32) *Puppy
+	UpdatePuppy(ID uint32, pet *Puppy)
+	DeletePuppy(ID uint32) bool
+}
+
+type MapStore struct {
+	m map[uint32]Puppy
+}
+
+func CreateStore() *MapStore {
+	return &MapStore{make(map[uint32]Puppy)}
+}
+
+func (ms *MapStore) CreatePuppy(puppy *Puppy) {
```

Comment:
Puppy is small, you can work without pointers as it's being copied on map insertion (better double check that I'm saying the right thing here ;) )